daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ValueTranslator.scala

```diff
@@ -51,6 +51,7 @@ private[engine] class ValueTranslator(compiledPackages: CompiledPackages) {
 
       def go(typ: Type): Type =
         typ match {
+          case TSynApp(_, _) => throw new RuntimeException("TODO #3616,ValueTranslator")
```

Comment:
`ValueTranslator` translates incoming Ledger API values to speedy expressions so it only has to handle serializable types. Therefore, you can just error out here using `fail` like we do for `TForall` and `TStruct`.

Reply:
@cocreature is right. Just handle this case as `TForall` and `TStruct`.

Reply:
ok, done